pytorch_lightning/loops/fit_loop.py

```diff
@@ -173,6 +174,11 @@ def run(self) -> None:
 
     def on_run_start(self) -> None:
         """Calls the ``on_train_start`` hook."""
+        # create progress tracker
+        optimizations = tuple(OptimizationProgress() for _ in self.trainer.optimizers)
```

Comment:
The dataclasses should handle this, would just need to pass the number of optimizers

Reply:
Moved to dataclass.